docs/getting-started/roles/index.rst

```diff
@@ -0,0 +1,67 @@
+==================================
+Built-in Role-Based Access Control
+==================================
+
+oso includes a library that makes it easy to add roles to your application.
+It integrates with our model library integrations and allows you to create role
+relationships between your user models and your resource models.
+We then generate some polar to make it easy to define rules over these roles instead
+of directly over your users and provide helper methods for managing roles.
+
+This feature is currently supported in the SQLAlchemy library.
+
+
+If you want to get started with SQLAlchemy roles, look at
+:doc:`/getting-started/roles/sqlalchemy/basic_roles`.
+
+For a more in-depth understanding of roles, check out our :doc:`guide to
+Role-Based Access Control (RBAC) </getting-started/roles/rbac>`.
+
+.. toctree::
+    :hidden:
+    :maxdepth: 1
+
+    Guide to RBAC <rbac>
+    sqlalchemy/basic_roles
+
+
+How it works
+============
+
+Lets say you have a User class and a Widget class and you want to assign users a role for a widget like
+"OWNER" or "User". Using the roles library you can generate a WidgetRole model which allows you to assign
+a user a role for a Widget. The schema for this new model's table looks like this.
+
+.. image:: /getting-started/builtin-roles/roles.svg
+
+It's a join table between
+User and Widget that contains some metadata like the name of the role. In SQLAlchemy library we add User.widgets
```

Comment:
We should probably make all the things code literals here. \`\`User\`\`, \`\`User.widgets\`\`, etc.